core-web/src/main/java/org/silverpeas/core/webapi/attachment/SimpleDocumentListResource.java

```diff
@@ -65,10 +67,36 @@ public String getResourceId() {
     List<SimpleDocument> docs =
         AttachmentServiceProvider.getAttachmentService().listDocumentsByForeignKeyAndType(
             new ForeignPK(getResourceId(), getComponentId()), DocumentType.attachment, lang);
+    return asWebEntities(docs);
+  }
+
+  /**
+   * Returns documents of specified resource in the specified language
+   *
+   * @param lang the wanted language.
+   * @return documents of specified resource
+   */
+  @GET
+  @Path("types/{type}/{lang}")
+  @Produces(MediaType.APPLICATION_JSON)
+  public List<SimpleDocumentEntity> getDocumentsByType(@PathParam("type") final String type,
+      @PathParam("lang") final String lang) {
+    DocumentType documentType = DocumentType.decode(type);
+    if (documentType == null) {
+      return Collections.emptyList();
+    }
+    List<SimpleDocument> docs =
+        AttachmentServiceProvider.getAttachmentService().listDocumentsByForeignKeyAndType(
+            new ForeignPK(getResourceId(), getComponentId()), documentType, lang);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "ForeignPK"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)